Sources/SwiftParser/Parameters.swift

```diff
@@ -156,7 +156,22 @@ extension Parser {
       defaultValue = nil
     }
 
-    let trailingComma = self.consume(if: .comma)
+    var trailingComma: Token?
+    if self.at(.comma) {
+      trailingComma = self.consume(if: .comma)
+    } else if !self.at(.rightParen) {
+      let canParseIdentifier: Bool = withLookahead {
+        $0.canParseTypeIdentifier(allowKeyword: false)
+      }
+
+      let canParseAttribute: Bool = withLookahead {
+        ($0.consume(if: .atSign) != nil && $0.canParseCustomAttribute())
+      }
+
+      if canParseIdentifier || canParseAttribute {
+        trailingComma = Token(missing: .comma, arena: self.arena)
+      }
```

Comment:
Not sure if there is any noteworthy performance hit here? 
We have two `withLookahead` where we potentially only use `canParseIdentifier`. 

Was considering 

```swift
let canParseIdentifier: Bool = withLookahead {
  $0.canParseTypeIdentifier(allowKeyword: false)
}

if canParseIdentifier {
  trailingComma = Token(missing: .comma, arena: self.arena)
}

let canParseAttribute: Bool = withLookahead {
  $0.consume(if: .atSign) != nil && $0.canParseCustomAttribute()
}

if canParseAttribute {
  trailingComma = Token(missing: .comma, arena: self.arena)
}
```

Reply:
I think your current implementation is good. This will only be hit for invalid code and we tend to not squeeze the last bit of performance for invalid code because it’s A LOT less common than valid code.